pkg/api/types.go

```diff
@@ -584,10 +584,28 @@ const (
 	ClaimLost PersistentVolumeClaimPhase = "Lost"
 )
 
+type HostPathType string
+
+const (
+	// If nothing exists at the given path, an empty directory will be created there. Otherwise, behaves like HostPathExists
+	HostPathUnset HostPathType = ""
+	// If nothing exists at the given path, the pod will fail to run and provide an informative error message
+	HostPathExists HostPathType = "exists"
```

Comment:
Constant values start with caps - "Exists".

Drop the " and provide an informative error message" - that's assumed.  In fact, I would restate these comments in the affirmative.  "A directory entry of any type must exist at the given path".